### PR TITLE
Bugfix/sdk switch integration and e2e tests

### PR DESCRIPTION
**Pull request by weilbith**
Related to #1584

**Short description**
How to explain this shortly... As I talked with Andre is has turned out that the integration and end-to-end tests of the SDK have evolved awkwardly over the time. So are today the integration tests actually the end-to-end tests and vice-versa. This leads to confusion when we now start to add end-to-end tests for the dApp. Also is the general handling like in the CI configuration confusing. The unit and e2e tests run in the default workflow. But the integration tests nightly. Read more in my commit message(s) for further details or examples.
After all Andre agreed that is makes sense to just switch both names. So the what is now called integration tests for the SDK become the e2e tests and the over way around. After all the tests and everything functional wise hasn't changed by this naming. But it allowed to clean up this awkward naming conflict that has evolved with the dApp tests now.
This PR includes mainly renaming. Adjusting docs and references according to it. But also small improvement so the bash script (using a linter) have been done, as well as a convenient npm script for the developer. **Please review this PR commit by commit**. Due to the limitations of how Git works this is pretty confusing else (moving, renaming files and change their content does not get presented well).

_MISSING_ is the renaming of the DockerHub repository. This is still called `raidennetwork/lightclient-integration`. According to the new defined naming it should be a `raidennetwork/lightclient-e2e-environment`. Also do we need to add a tagging system for this image, since CircleCI is about to restrict `latest` pulls.

**Definition of Done**

- [ ] Steps to manually test the change have been documented
- [ ] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1.
2.

### CODE REVIEW

File: raiden-ts/tests/e2e/e2e.spec.ts [hunk 1]
```diff
@@ -55,24 +55,24 @@ async function createRaiden(account: number | string | Signer): Promise<Raiden>
 }
 
 const wait = async (timeInMs: number): Promise<void> =>
-  new Promise((resolve) => setTimeout(() => resolve(), timeInMs));
```
Comment (andrevmatos) [inline raiden-ts/tests/e2e/e2e.spec.ts]: Nitpick: these parenthesis changes are against our current linter rules. Do you have up-to-date packages?

File: raiden-ts/tests/e2e/e2e.spec.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ async function createRaiden(account: number | string | Signer): Promise<Raiden>
 }
 
 const wait = async (timeInMs: number): Promise<void> =>
-  new Promise(resolve => setTimeout(() => resolve(), timeInMs));
```
Comment (andrevmatos) [inline raiden-ts/tests/e2e/e2e.spec.ts]: Oh, it's done already. Thanks

**Review by andrevmatos — APPROVED** [review]
lgtm, thank you

**Comment by weilbith** [comment]
I'm so ignorant and disregard what CodeClimate is reporting. It is quite picky and not related to this PR. Apparently it was once accepted like this. If it pops-up again we might handle it.

**Pull request merged**